src/shipping/views/WeightRatesUpdate/WeightRatesUpdate.tsx

```diff
@@ -251,14 +266,23 @@ export const WeightRatesUpdate: React.FC<WeightRatesUpdateProps> = ({
 
   const onPostalCodeUnassign = code => {
     if (code.id !== undefined) {
-      setCodesToDelete([...codesToDelete, code.id]);
-      rate.postalCodeRules = rate.postalCodeRules.filter(
-        rule => rule.id !== code.id
-      );
+      dispatch({
+        updateStateProps: {
+          codesToDelete: [...state.codesToDelete, code.id],
+          havePostalCodesChanged: true,
+          postalCodeRules: state.postalCodeRules.filter(
+            rule => rule.id !== code.id
```

Comment:
Since we use this condition in quite some places in the project I'd find the `getById` and add a corresponding getByUnmatchingId or something and use it here